message db_role_info 
{
required int32 role_id = 1;
required int32 uid = 2;
required int32 tid = 3;
required string name = 4;
required int32 gender = 5;
required int32 job_type = 6;
required int32 role_level = 7;
required int32 role_exp = 8;
required int32 costume_tid = 9;
required int32 title = 10;
required int32 coin = 11;
required int32 coupon = 12;
required int32 diamond = 13;
required int32 vip_end_ts = 14;
required int32 vip_level = 15;
required int32 map_id = 16;
required int32 map_x = 17;
required int32 map_y = 18;
required int32 last_map_id = 19;
required int32 last_map_x = 20;
required int32 last_map_y = 21;
required int32 flag = 22;
required int32 status = 23;
required int32 server_id = 24;
required int32 create_time = 25;
required int32 team_id = 26;
required int32 team_create_time = 27;
required int32 bag_size = 28;
required int32 depot_size = 29;
required int32 buddy_size = 30;
required int32 lineup_size = 31;
required int32 default_lineup_id = 32;
required int32 lineup_capacity = 33;
required int32 login_time = 34;
required int32 last_logoff_time = 35;
required string login_ip = 36;
required int32 instance_count = 37;
required int32 instance_time = 38;
required int32 guild_id = 39;
required int32 guild_time = 40;
required string guild_name = 41;
required int32 phy_strength = 42;
required int32 extra_phy_strength = 43;
required int32 time_to_recover_phy_strength = 44;
required int32 time_to_recover_extra_phy_strength = 45;
required int32 renown = 46;
}
